### PR TITLE
Specify correct branches to run CI workflow on commits and PRs

### LINKED ISSUE #484: Evaluate Github Actions CI
Set up [Github Actions](https://github.com/features/actions) for our CI use and try it for CD as well.

For CI, it's probably simple as we need only automate:

* build
* tests
* linting

`npm install`, `npm run build`, `npm run test` and `npm run lint` are the commands you would need for that.

If possible, during build of PRs it would be interesting to see if the GH Actions supports moving the associated issue to the `In Progress` column of the [Active Sprint project](https://github.com/orgs/creativecommons/projects/7).

### PR DESCRIPTION
**Pull request by SaurabhAgarwala**
## Fixes

Fixes #484 by @brenoferreira 

## Description

The CI workflow is currently configured to run on PRs and commits on the `master` branch. But for this repo, the PRs are merged to `develop` branch. Hence this PR specifies the correct branches to run the CI workflow on commits and PRs.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by SaurabhAgarwala** [comment]
Finally the tests passed, after a lot of debugging. Please review it @brenoferreira

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,23 +23,16 @@ jobs:
...
       uses: actions/setup-node@v1
       with:
-        node-version: '12.16.1'
-
-    - id: cache-node-modules
-      name: Cache Node.js modules
-      uses: actions/cache@v1
-      with:
-        path: ~/.npm  # npm caches files in ~/.npm
-        key: node-${{ hashFiles('**/package-lock.json') }}
```
Comment (brenoferreira) [inline .github/workflows/ci.yml:32]: I think these things can be kept

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -4,9 +4,10 @@ on:
...
     branches:
     - master
+    - develop
   pull_request:
     branches:
-    - master
+    - develop
 
 env:
   PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: true
```
Comment (brenoferreira) [inline .github/workflows/ci.yml:13]: We don't need this env variable

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -22,23 +23,16 @@ jobs:
...
-    - id: cache-node-modules
-      name: Cache Node.js modules
-      uses: actions/cache@v1
-      with:
-        path: ~/.npm  # npm caches files in ~/.npm
-        key: node-${{ hashFiles('**/package-lock.json') }}
+        node-version: '12.14.1'
 
     - name: Install dependencies
-      run: npm install --prefer-offline --no-audit
```
Comment (brenoferreira) [inline .github/workflows/ci.yml:35]: You can revert this

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -22,23 +23,16 @@ jobs:
     - name: Setup Node.js
       uses: actions/setup-node@v1
       with:
-        node-version: '12.16.1'
```
Comment (brenoferreira) [inline .github/workflows/ci.yml:25]: Also, it seems like this version of Node has a problem with our setup which causes the build to fail. Can you use `v12.4.0`?

**Comment by SaurabhAgarwala** [comment]
> Thanks for your contribution.
> 
> I just made a few changes here to improve the CI workflow

**Comment by SaurabhAgarwala** [comment]
Sorry, I missed the changes requested message and couldn't update the changes. Thanks for updating and merging the PR

**Pull request merged**